### ISSUE TITLE
Check kube-proxy, and all other "kube-system" pods health in Vagrant validator.

### ISSUE THREAD

**Issue by jayunit100** [kubernetes/kubernetes#17792]
The vagrant recipes spin up clusters and verify scheduler, controller manager.

**Background**
The current vagrant recipes use a containerized salt stack that pulls containers for various core components, like the proxy itself.  Some of these containers might fail pulling for a variety of reasons, even though the other services (etcd, controller-manager, ...) are working.  

The trend towards "total bootstrapping" of kubernetes and the increasing importance of add-on / system containers, means that we will be in a rat-race to keep up with the amount of stuff you need to verify in a clutser.

**We should just make vagrant validate at the same granularity as E2Es**

Possibly a better / complimentary validation to run after "up" on a new cluster: would be the same one which e2e tests run: Making sure that all the pods in `kube-system` are transitioned into `Running` state before starting. .  This should hep us bring the vagrant validation to parity without having to continually maintain it, since introspecting what needs to run in kube-system is an automated task.

tl;dr  this is just proposing we add validation of all kube-system pods to be in "Running" state to the cluster validation.

**Comment by davidopp** [comment]
@mikedanese for his opinion

**Comment by jayunit100** [comment]
cc this will catch issues like #17797
